File: Python/SelectionSort.py
def SelectionSort(arr):
    for i in range(size-1):
        min =i
        for j in range(i+1, size):
            if arr[j]<arr[min]:
                min = j
            if min != i:
                arr[min], arr[i] = arr[i], arr[min]
    return arr

try:
    arr = []
    size = int(input("Enter the size of the list: "))
    for _ in range(size):
        while True:
            try:
                el = int(input("Enter a number: "))
                arr.append(el)
                break 
            except ValueError:
                print("Invalid input. Please enter a valid number.")

    print("INPUT LIST:", arr)
    SelectionSort(arr)
    print("SORTED LIST:", arr)
except KeyboardInterrupt:
    print("Program interrupted by the user.")
except Exception as e:
    print(f"INPUT MUST BE IN NUMBER")

File: Javascript/SelectionSort.js
function SelectionSort(arr) {
    for (let i = 0; i < arr.length - 1; i++) {
        let min = i;
        for (j = i + 1; j < arr.length; j++) if (arr[j] < arr[min]) min = j;
        if (min != j) [arr[i], arr[min]] = [arr[min], arr[i]];
    }
}

let arr = [5, 4, 3, 2, 1];
console.log("ARRAY BEFORE SORTING:");
console.log(arr);
SelectionSort(arr);
console.log("ARRAY AFTER SORTING:");
console.log(arr);
